diff --git a/node_modules/oicq/lib/core/device.js b/node_modules/oicq/lib/core/device.js
index ffadc63..e6c434d 100644
--- a/node_modules/oicq/lib/core/device.js
+++ b/node_modules/oicq/lib/core/device.js
@@ -34,7 +34,7 @@ function generateImei(uin) {
     return imei + calcSP(imei);
 }
 /** 生成短设备信息 */
-function generateShortDevice(uin) {
+function generateShortDeviceDeprecated(uin) {
     const hash = (0, constants_1.md5)(String(uin));
     const hex = hash.toString("hex");
     return {
@@ -56,6 +56,45 @@ function generateShortDevice(uin) {
         "--end--": "修改后可能需要重新验证设备",
     };
 }
+/** 生成随机设备信息 */
+function generateShortDevice(uin) {
+    function randomString(length) {
+        const pool = "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789";
+        let result = "";
+
+        for (let i = 0; i < length; i++) {
+            result += pool.charAt(Math.floor(Math.random() * pool.length));
+        }
+
+        return result;
+    }
+
+    function randomInt(boundary) {
+      return Math.floor(Math.random() * boundary);
+    }
+
+    return {
+        "--begin--": "该设备由账号作为seed固定生成，账号不变则永远相同",
+        "--notice--": "The code that generated this file was patched by Adachi-BOT",
+        product: `ILPP-${randomString(5).toUpperCase()}`,
+        device: `${randomString(5).toUpperCase()}`,
+        board: `${randomString(5).toUpperCase()}`,
+        brand: `${randomString(4).toUpperCase()}`,
+        model: `ILPP ${randomString(4).toUpperCase()}`,
+        wifi_ssid: `HUAWEI-${randomString(7)}`,
+        bootloader: "U-boot",
+        android_id: `IL.${randomInt(10000000)}.${randomInt(10000)}`,
+        boot_id: `${randomString(8)}-${randomString(4)}-${randomString(4)}-${randomString(4)}-${randomString(12)}`,
+        proc_version: `Linux version 5.10.101-android12-${randomString(8)}`,
+        mac_address: `2D:${randomString(2).toUpperCase()}:${randomString(2).toUpperCase()}:${randomString(
+            2
+        ).toUpperCase()}:${randomString(2).toUpperCase()}:${randomString(2).toUpperCase()}`,
+        ip_address: `192.168.${randomInt(255)}.${randomInt(255)}`,
+        imei: `86${randomInt(10000000000000)}`,
+        incremental: `${randomString(10).toUpperCase()}`,
+        "--end--": "修改后可能需要重新验证设备。",
+    };
+}
 exports.generateShortDevice = generateShortDevice;
 /** 生成完整设备信息 */
 function generateFullDevice(d) {
@@ -103,18 +142,21 @@ var Platform;
     Platform[Platform["iMac"] = 4] = "iMac";
     Platform[Platform["iPad"] = 5] = "iPad";
 })(Platform = exports.Platform || (exports.Platform = {}));
+// XXX Here update device informations based on mirai
+// https://github.com/mamoe/mirai/blob/e7bcb190a7662350211897b29314bad0f332e6d9/mirai-core/src/commonMain/kotlin/utils/MiraiProtocolInternal.kt
 const mobile = {
     id: "com.tencent.mobileqq",
-    name: "A8.8.80.7400",
-    version: "8.8.80.7400",
-    ver: "8.8.80",
-    sign: Buffer.from([166, 183, 69, 191, 36, 162, 194, 119, 82, 119, 22, 246, 243, 110, 182, 141]),
+    name: "A8.9.23.9425",
+    version: "8.9.23.9425",
+    ver: "8.9.23",
+    sign: Buffer.from([0xA6, 0xB7, 0x45, 0xBF, 0x24, 0xA2, 0xC2, 0x77, 0x52, 0x77, 0x16, 0xF6, 0xF3, 0x6E, 0xB6, 0x8D]),
     buildtime: 1640921786,
-    appid: 16,
-    subid: 537113159,
-    bitmap: 184024956,
-    sigmap: 34869472,
-    sdkver: "6.0.0.2494",
+    appid: 19,
+    subid: 537143097,
+    bitmap: 150470524,
+    sigmap: 16724722,
+    subsigmap: 0x10400,
+    sdkver: "6.0.0.2530",
     display: "Android",
 };
 const watch = {
